extensions/hadoop/src/main/java/com/hazelcast/jet/hadoop/impl/ReadHadoopNewApiP.java

```diff
@@ -206,7 +214,7 @@ private Supplier(
         ) {
             this.configuration = configuration;
             this.projectionFn = projectionFn;
-            this.assignedSplits = assignedSplits;
+            this.assignedSplits = reAssignSplits(assignedSplits);
```

Comment:
This still happens on the coordinator node, the reassign must be in `get(int)` method.

Reply:
👍